src/main/java/com/money_account_service/utility/RequestInterceptor.java

```diff
@@ -0,0 +1,26 @@
+package com.money_account_service.utility;
+
+import com.money_account_service.dtos.response.AuthorizeResponseDto;
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Optional;
+
+@AllArgsConstructor
+public class RequestInterceptor {
+
+    private UserServiceClient userServiceClient;
+
+    public Optional<AuthorizeResponseDto> authorizeRequest(HttpServletRequest requestServlet)
```

Comment:
We never return `Optional.empty()`, because we throw exception otherwise. We can simplify the return type.

Also, I think `AuthorizeResponseDto` should not leak outside of the `userServiceClient`. We should return some domain user representation for the use of our application.